### ISSUE TITLE
compiler error in rapidjson/document.h

### ISSUE THREAD

**Issue by maxaehle** [cern-fts/davix#130]
When I try to compile davix as part of ROOT, I get the following error message related to [document.h:319](https://github.com/cern-fts/davix/blame/3f08d4d2a460e2edce04c53daa01c78efbb7aa1c/src/libs/rapidjson/document.h#L319):

```
/..../root/build/builtins/davix/DAVIX-prefix/src/DAVIX/src/libs/rapidjson/document.h: In member function ‘rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)’:
/..../root/build/builtins/davix/DAVIX-prefix/src/DAVIX/src/libs/rapidjson/document.h:319:82: error: assignment of read-only member ‘rapidjson::GenericStringRef<CharType>::length’
  319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
      |                                                                           ~~~~~~~^~~~~~~~~~~~
```

I agree that the line looks strange, `length` is a `const SizeType` member and should not be assigned. Also, I wonder if the missing return statement could create bugs.

**Comment by maxaehle** [comment]
the error comes up with GCC 14.2.0, but not with GCC 8.5.0

**Comment by maxaehle** [comment]
GCC 13.2.0 compiles the code as well.

The issue is related to an out-of-date version of RapidJSON. This has partially been addressed in #120, allowing to specify a more up to date version of RapidJSON.

**Comment by mpatrascoiu** [comment]
Closed by pull-request #129 (`Remove non-compiling assignment operator`)

Further tracking will be done here: [DMC-1426](https://its.cern.ch/jira/browse/DMC-1426) / [DMC-1427](https://its.cern.ch/jira/browse/DMC-1427)

Given there are a couple of things pending in the current `devel` branch, I think I'll do a Davix `R_0_8_8` release this side of the New Year.

Cheers,
Mihai
